app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -58,6 +59,10 @@ def requested_authn_contexts
                                   [default_aal_context]
   end
 
+  def requested_authn_context_comparison
+    @requested_authn_context_comparison ||= saml_request.requested_authn_context_comparison
```

Comment:
no point in memoizing this IMO, especially since the value can be nil, I think we could just inline the method

Reply:
Makes sense, was mostly following the patterns in the rest of the class. I'll update this.